index.js

```diff
@@ -61,7 +61,10 @@ function formatOpts(opts) {
   opts.key = opts.key || 'koa:sess';
 
   // back-compat maxage
-  if (!('maxAge' in opts)) opts.maxAge = opts.maxage;
+  if ('maxAge' in opts) {
+    const validMaxAge = opts.maxAge === 'session' || typeof opts.maxAge === 'number';
+    if (!validMaxAge) throw new TypeError('opts.maxAge must be a number or string `session`');
+  } else opts.maxAge = opts.maxage;
```

Comment:
it does not check opts.maxage here. I think we should keep the original logic, and check opts.maxAge after `if (!('maxAge' in opts)) opts.maxAge = opts.maxage;`

Reply:
it's actually using the origin logic? just adding a case for when there is maxAge.

Reply:
do you mean you want it this way?

if (!('maxAge' in opts)) opts.maxAge = opts.maxage;
else {
    const validMaxAge = opts.maxAge === 'session' || typeof opts.maxAge === 'number';
    if (!validMaxAge) throw new TypeError('opts.maxAge must be a number or string `session`');
}

Reply:
```js
if (!('maxAge' in opts)) opts.maxAge = opts.maxage;
const validMaxAge = opts.maxAge === 'session' || typeof opts.maxAge === 'number';
if (!validMaxAge) throw new TypeError('opts.maxAge must be a number or string `session`');
```

Reply:
and I think if we add this validation, somebody's code may be broken and this will be a breaking change